### DISCUSSION TITLE
Troubleshooting Schema

### DISCUSSION THREAD

**Discussion by kirkilj** [jelovirt/org.lwdita#165] [General]
@jelovirt, Is there a plan to provide a built-in schema for the DITA Troubleshooting DOCTYPE, or that an exercise left to the student?

**Comment by jelovirt** [comment]
No plans at this time.

**Reply by kirkilj** [reply]
Likewise, adding {.troubleshooting} generates <topic outputclass="troubleshooting"> by treating it as a regular attribute list item.

The Troubleshooting topic schema is kind of weird in that you wouldn't know in which element should be used to wrap the markdown content. It can include one or more task nested topics and your ordered list to steps conversion could handle that aspect, but the rest is a crap shoot. I suppose it would be possible to invoke an XML Transformer on the DITA with attribute-list generated outputclasses to turn them into elements and change the DOCTYPE, but that's beyond me at the current time. Perhaps you see a utility in having a Transformer phase as an option in the conversion process for use cases you're privy to.
